imap_processing/tests/hi/test_hi_l1c.py

```diff
@@ -325,6 +328,102 @@ def test_get_tof_window_mask():
     np.testing.assert_array_equal(expected_mask, window_mask)
 
 
+def test_empty_pset_dataset_arbitrary_cal_prod_numbers(use_fake_repoint_data_for_time):
+    """Test empty_pset_dataset with non-sequential calibration product numbers."""
+    n_energy_steps = 3
+    l1b_esa_energy_steps = xr.DataArray(
+        data=np.concat((np.arange(n_energy_steps + 1).repeat(2), np.array([255, 255]))),
+        attrs={"FILLVAL": 255},
+    )
+    # Use non-sequential calibration product numbers
+    cal_prod_numbers = np.array([5, 10, 100])
```

Comment:
Put these out of order as well to try and test the sorting is done as you expect?
```suggestion
    cal_prod_numbers = np.array([10, 5, 100])
```

Reply:
Oh, actually this change results in non-sorted calibration product because the sorting is done outside of the `empty_pset_dataset()` function.